### PR TITLE
Make this repository an Ember Addon

### PR DESCRIPTION
**Pull request by lukemelia**
- [x] move boxel components to addon dir
- [x] Get dummy app working with freestyle.
- [x] Setup usage-support for assets consumed by usage components.
- [x] colocated component css works for addon and for dummy app 
- [x] Strip usage files from non-dummy builds
- [x] Make sure images load
- [x] Fix freestyle source parsing https://github.com/chrislopresto/ember-freestyle/issues/481
- [x] Documentation
- [x] Tests passing on CI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
+
+Boxel component implementations can be found in the directory `addon/components/boxel`. Most components have a `usage` component colocated  with the component implementation. These usage components are included in the living component explorer at the `/docs` route in the dummy app. The usage files are stripped out of builds non-dummy-app builds, so they are not included in your app's payload.
+
+The component implementations are automatically re-exported to make them available to consuming apps. In most Ember addons, you would see files in the app directory to perform these re-exports. This addon generates these files as part of its build, so you won't see them in the repository.
+
+Component implementations also each have a colocated CSS file that is included in the addon's CSS and becomes part of the consuming app's vendor.css file.
+
+The dummy app is built using embroider.
+
+In addition to the Docs section, the dummy app also includes a few scenarios, which use the components in the context of a real-world example.
```
Comment (mattmcmanus) [inline README.md]:
```suggestion
The repository provides an Ember Addon allowing usage of the Ember component library for the Boxel Design System.

It consists of several parts:
1. The Boxel Design System Components and their documentation
2. Various prototypes using the design system
3. Miscellaneous components that will eventually be moved

## Design system components
All components in `/addon/components` will be used in whatever apps consume the design system. Each component folder consists of several parts:
- `index.hbs` & `index.js`: The template and javascript for the component
    - *Note:* Unlike a usual Ember addon, these files are automatically reexported in whatever app consumes the addon. So there is no need to handle that on your own.
- `index.css`: The pure CSS styles associated with the component. This file will automatically be added to the consuming app as well
- `usage.hbs` & `usage.js`: A standardized way to document and experiment with the designs of each component. All usage files will automatically show up in the components list in `/docs` in the dummy app. Utilizes chrislopresto/ember-freestyle
    - *Note:* These files are automatically stripped out of any production build

## Prototypes
The prototypes are there to exercise in context various aspects of the design system. 

## Miscellaneous Components
There are many components that are specific to a certain prototype or will eventually be moved to other repos, like the main Cardhost app. 
```

**Review by mattmcmanus — CHANGES_REQUESTED** [review]
I left a stab at the documentation. 

It'd also be great to rewrite the commit history to reflect the story more accurately. I'm happy to take a stab at that.

**Comment by lukemelia** [comment]
>It'd also be great to rewrite the commit history to reflect the story more accurately. I'm happy to take a stab at that.

@mattmcmanus go fot it

**Review by mattmcmanus — APPROVED** [review]

**Comment by mattmcmanus** [comment]
@lukemelia Updated

**Review by burieberry — APPROVED** [review]

**Pull request merged**